example/12_reduce/reduce_blockwise.cpp

```diff
@@ -13,21 +13,21 @@
 #include "device_base.hpp"
 #include "device_reduce_blockwise.hpp"
 #include "host_reduce_util.hpp"
-#include "host_generic_reduction.hpp"
+#include "host_reduction.hpp"
 
 #include "reduction_enums.hpp"
 #include "reduction_operator_mapping.hpp"
 
 using namespace ck;
 using namespace ck::tensor_operation::device;
 
-using InDataType  = half_float::half;
-using OutDataType = half_float::half;
+using InDataType  = ck::half_t;
+using OutDataType = ck::half_t;
 using AccDataType = float;
 
-using kInDataType  = ck::half_t;
-using kOutDataType = ck::half_t;
-using kAccDataType = float;
+using hInDataType  = half_float::half;
```

Comment:
Something wrong with using ck::half_t on host?

Reply:
The reason is that Reduction needs to use `abs()` and `isnan()` on fp16.  But for `ck::half_t`,  the `__habs()` and `__hisnan()` can only be used in `__device__` mode to do the functionality of `abs()` and `isnan()`.  In the other side,  half_float::half has direct and complete implementation of abs() and isnan() on the host side.